test/integration/scheduler/filters/filters_test.go

```diff
@@ -814,19 +799,537 @@ func TestInterPodAffinity(t *testing.T) {
 			pods: []*v1.Pod{
 				{Spec: v1.PodSpec{
 					Containers: []v1.Container{{Name: "container", Image: imageutils.GetPauseImageName()}},
-					NodeName:   nodes[0].Name}, ObjectMeta: metav1.ObjectMeta{Name: "fakename", Labels: map[string]string{"foo": "abc"}}},
+					NodeName:   "testnode-0"}, ObjectMeta: metav1.ObjectMeta{Name: "fakename", Labels: map[string]string{"foo": "abc"}}},
 			},
 			fits: false,
 		},
+		{
+			name: "anti affinity: matchLabelKeys ANDed with LabelSelector when LabelSelector isn't empty (feature flag: enabled)",
```

Comment:
(not related to this line)

You craft a lot of integration tests, which is definitely good :) However, there is 0 new UT in pkg/scheduler/framework/plugins/interpodaffinity. If you hadn't started building the integration tests, I would suggest moving some of them to UTs as UTs are more lightweight and achieve the same goal. But given we're here already, could you add some non-overlapping UTs (with current integration tests)? (I noticed minDomains and matchLabels feature did have some UTs in podtopologyspread pkg)

Reply:
> could you add some non-overlapping UTs (with current integration tests)?

I cannot think of other cases apart from the ones in integration tests.

and...

> as UTs are more lightweight and achieve the same goal.

this is 100% true... 
So, I’d say we can even move all cases to unit tests. (but I’m thinking we can keep this as follow-up task as these integration tests has the same values to check the functionalities)
I’m now wondering which test case we want to move to unit tests, and which test case we can keep in the integration tests.

Reply:
sgtm, you can mark it as a follow-up item in beta.

Reply:
+1 way too many integration tests. I suggest having just one case for affinity and one for antiaffinity.